src/hAPI/README.md

```diff
@@ -1,5 +1,54 @@
+* Click the "Connect To Haply" button and select the COM port for the Haply device. Steadily grasp the 2DIY's knob and click the "START" button to initiate the audio-haptic experience.
+* The audio-haptic experience is equivalent to that shown in state diagram below, i.e., it plays a TTS narration for each segment, followed by a sonified experience of all subsegments within that segment, and then a haptically guided tour of each subsegment with the 2DIY. Once all segments are covered, the same process is repeated for objects, starting with grouped objects and then ungrouped objects. Subsegments are defined as isolated contours of a segment, as shown in the image below.
+* The Previous, Next, and Stop buttons can be used to navigate back and forth in the interaction, which always follows the pattern:
+![a state machine diagram explaining how the audio-haptic experience works](./images/haptics_state_machine.png){:height="250px"}
+NOTE:  this diagram will be changed slighly to indicate a starting state
+
+![differentiating an object, segment, and subsegment](./images/object_segments.png){:height="250px"}
+
+ For example, the clicking the "Next Button" during the TTS narration of a segment will immediately skip the TTS and move to the sonified experience of that segment. Clicking the "Next" button again will move to the guided tour of that segment, starting with the first subsegment. Clicking "Previous" during a TTS narration of the second segment will run a guided tour of the last subsegment of the first segment.
+
+## How does it work?  
+
+![the dataflow between the main browser script, the worker, and the 2DIY](./images/browser_arch.png)
+
+Data from the handler is sent to the ```hapi-utils.ts``` main script file.
+
+Code to play audio files is contained within the main script, while all haptic code runs inside a web worker. Since the audio and haptic experiences are intended to be run one after the other in a timed-fashion, there is back and forth communication between the main script and worker to keep track of state variables. For example, the ```doneWithAudio``` variable is set to true once an audio file has finished playing, sent to the worker via a ```postMessage()```, and  then acknowledged by the worker to ready the next haptic segment or object to be played. The code that handles the interactivity with the Previous, Next, and Stop buttons is also contained in the ```worker.ts``` file.
+
+## Haptic Loop
+
+Code for the haptic calculations is contained in the ```worker.ts``` webworker file. A web worker allows scripts to be executed in background threads, allowing it to perform tasks without interfering with the main script. Since the force calculations required for the 2DIY can be intensive, the web worker ensures that the performance of the main script is not hindered.
```

Comment:
Note that you're assuming the reader knows what a haptic loop generally is (e.g., a periodic function with strict timing constraints that needs to quickly update the I/O of the physics simulation and render new values back to the haptic device). How the web worker "ensures that the performance [...] is not hindered" may not be clear to a novice.

Reply:
I slightly rewrote the 'how does it work' and 'haptic loop' sections, but I'm not sure if they read better now.